router/batchrouter/batchrouter.go

```diff
@@ -1809,12 +1809,11 @@ func (worker *workerT) workerProcess() {
 						brt.recordUploadStats(*batchJobs.BatchDestination, output)
 					}
 
-					destUploadStat.End()
 				case misc.Contains(warehouseutils.WarehouseDestinations, brt.destType):
 					useRudderStorage := misc.IsConfiguredToUseRudderObjectStorage(batchJobs.BatchDestination.Destination.Config)
 					objectStorageType := warehouseutils.ObjectStorageType(brt.destType, batchJobs.BatchDestination.Destination.Config, useRudderStorage)
 					destUploadStat := stats.Default.NewStat(fmt.Sprintf(`batch_router.%s_%s_dest_upload_time`, brt.destType, objectStorageType), stats.TimerType)
-					destUploadStat.Start()
+					defer destUploadStat.RecordDuration()()
```

Comment:
it would me more appropriate to not use a defer here and clearly define the timer's start and end
```
start := time.Now()
...
stat.Since(start)
```